docs/reference/modules/discovery/discovery.asciidoc

```diff
@@ -62,22 +62,38 @@ logs. You must fix the master node's instability before addressing these other
-meaningless. The cluster metadata is stored on a subset of the master-eligible
-nodes in the cluster. If a quorum cannot be discovered then the missing nodes
-were the ones holding the cluster metadata. If you cannot bring the missing
-nodes back into the cluster, start a new cluster and restore data from a recent
-snapshot. Refer to <<modules-discovery-quorums>> for more information.
-
-The logs from the `ClusterFormationFailureHelper` may also indicate that it has
-discovered a possible quorum of master-eligible nodes. If so, the usual reason
-that the cluster cannot elect a master is that one of the other nodes cannot
-discover a quorum. Inspect the logs on the other master-eligible nodes and
-ensure that every node has discovered a quorum.
+On master-eligible nodes, the logs from the `ClusterFormationFailureHelper` may
+indicate that a master election requires a certain set of nodes and that it has
+not discovered enough nodes to form a quorum. If so, you must address the
+reason preventing {es} from discovering the missing nodes. The missing nodes
+are needed to reconstruct the cluster metadata. Without the cluster metadata,
+the data in your cluster is meaningless. The cluster metadata is stored on a
+subset of the master-eligible nodes in the cluster. If a quorum cannot be
+discovered then the missing nodes were the ones holding the cluster metadata.
+Ensure there are enough nodes running to form a quorum and that every node can
+communicate with every other node over the network. {es} will report
+information about failures to form a cluster in its logs, and will report
+additional details about network connectivity there if the problems persist for
+more than a few minutes. If you cannot start enough nodes to form a quorum,
+start a new cluster and restore data from a recent snapshot. Refer to
+<<modules-discovery-quorums>> for more information.
+
+On master-eligible nodes, the logs from the `ClusterFormationFailureHelper` may
+also indicate that it has discovered a possible quorum of master-eligible
+nodes. If so, the usual reason that the cluster cannot elect a master is that
+one of the other nodes cannot discover a quorum. Inspect the logs on the other
+master-eligible nodes and ensure that every node has discovered a quorum.
+
+If your cluster has a stable master which some nodes cannot discover or join
+then these nodes will report many kinds of error to clients and in their logs.
+You must address the obstacles preventing these nodes from joining the cluster
+before addressing other issues. It will not be possible to solve any other
+issues reported by these nodes while they are not able to join the cluster. On
+these nodes, the logs from the `ClusterFormationFailureHelper` will indicate
+which master-eligible nodes they can discover. {es} will report information
+about failures to join a cluster in its logs, and will report additional
+details about network connectivity there if the problems persist for more than
+a few minutes.
```

Comment:
Hi @DaveCTurner! Thanks for this addition. I know that the first two paragraphs existed before you added this last paragraph, but I think that breaking those up and providing some more details to link those conditions could be easier to scan and read. Let me know what you think 🤔 

On master-eligible nodes, the `ClusterFormationFailureHelper` logger can indicate
conditions where a master node election couldn't occur:

Missing cluster metadata::
Cluster metadata is stored on a subset of the master-eligible nodes in a 
cluster. Without this metadata, the data in your cluster is meaningless. If the
`ClusterFormationFailureHelper` logger indicates that {es} couldn't discover 
enough of these nodes to form a quorum, then it's likely that the missing nodes
are the ones holding the cluster metadata.
+
Ensure that there are enough nodes running in your cluster to form a quorum, and
that every node can communicate with every other node over the network. {es}
will report information about failures to form a cluster in its logs, and will 
report additional details there about network connectivity if the problems 
persist for more than a few minutes. If you can't start enough nodes to form a 
quorum, start a new cluster and restore data from a recent snapshot. Refer to
<<modules-discovery-quorums>> for more information.

Unable to discover a quorum::
The `ClusterFormationFailureHelper` logger may also indicate that it
discovered a possible quorum of master-eligible nodes. The typical reason that
the cluster can't elect a master is that one of the other nodes couldn't
discover a quorum. Inspect the logs on the other master-eligible nodes and
ensure that every node has discovered a quorum.

If your cluster has a stable master node that some nodes can't discover or join,
then these nodes will report multiple errors in their logs and to clients. On
each of these nodes, the `ClusterFormationFailureHelper` logger indicates which
master-eligible nodes they can discover. Elasticsearch will report information
in its logs about failures to join a cluster, including additional details
about network connectivity if problems persist for more than a few minutes.
You must address the obstacles preventing these nodes from joining the cluster
before addressing other issues.

Reply:
I know what you mean, it's a bit of a big paragraph, but I don't think it makes sense to split it up like that. "Missing cluster metadata" and "unable to discover a quorum" are not mutually exclusive, indeed "unable to discover a quorum" is an indication that the cluster metadata is missing.

I've pushed a different approach in 40a5414d006. Not sure about the formatting there yet, this might need your help to get it looking right, but I think the structure is better at least.